crd-generator/src/main/java/io/strimzi/crdgenerator/ApiVersion.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.crdgenerator;
+
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.Short.parseShort;
+
+/**
+ * Represents the version of a Kubernetes API, for example {@code v1alpha1} or {@code v2}.
+ * These version numbers are comparable, so {@code v1alpha1 < v1beta1 < v1 < v2alpha1} etc.
+ */
+public class ApiVersion implements Comparable<ApiVersion> {
+
+    public static final Pattern PATTERN = Pattern.compile("v([0-9]+)((alpha|beta)([0-9]+))?");
+    public static final ApiVersion V1ALPHA1 = parse("v1alpha1");
+    public static final ApiVersion V1BETA1 = parse("v1beta1");
+    public static final ApiVersion V1 = parse("v1");
+
+    private final short major;
+    private final short ab;
+    private final short minor;
+
+    public ApiVersion(short major, short ab, short minor) {
+        if (major < 0 || ab < 0 || ab > 2 || minor < 0) {
+            throw new RuntimeException();
+        }
+        this.major = major;
+        this.ab = ab;
+        this.minor = minor;
+    }
+
+    private static Matcher matcher(String apiVersion) {
+        return PATTERN.matcher(apiVersion);
+    }
+
+    public static boolean isVersion(String apiVersion) {
+        return matcher(apiVersion).matches();
+    }
+
+    public static ApiVersion parse(String apiVersion) {
+        Matcher matcher = matcher(apiVersion);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException("Invalid version " + apiVersion);
+        }
+        short major = parseShort(matcher.group(1));
+        short ab;
+        short minor;
+        String alphaBeta = matcher.group(3);
+        if (matcher.groupCount() > 1 && alphaBeta != null) {
+            if ("alpha".equals(alphaBeta)) {
+                ab = 0;
+            } else if ("beta".equals(alphaBeta)) {
+                ab = 1;
+            } else {
+                throw new IllegalStateException(alphaBeta);
+            }
+            minor = parseShort(matcher.group(4));
+        } else {
+            ab = 2;
```

Comment:
can we think more about using an enum as Alpha, Beta and Stable instead of short values 0, 1 and 2 (that anyway could be the corresponding underlying values of the enum)

Reply:
There's no benefit to using an enum because these things are entirely internal to `ApiVersion`. The intention is that people use the String form (v1alpha1) rather than using a constructor which takes actual values. Using constants would actually make the `compareTo` code harder to read imho, but I'll do it anyway.

Reply:
Well, I don't think that the benefit has to be for people using it only but even for developers maintaining it. I got the values now or at least you could document them so no need to update but I think that using enums don't hurt.